libs/fundamental-redirects/index.js

```diff
@@ -45,6 +47,46 @@ function externalRedirect(pattern, template, options = {}) {
   });
 }
 
+const fixableLocales = new Map();
+for (const locale of VALID_LOCALES.keys()) {
+  if (locale.includes("-")) {
+    // E.g. `en-US` becomes alias `en_US`
+    fixableLocales.set(locale.replace("-", "_").toLowerCase(), locale);
+  } else {
+    // E.g. `fr` becomes alias `fr-XX`
+    fixableLocales.set(`${locale}-\\w{2}`.toLowerCase(), locale);
+  }
+}
+
+for (const [alias, correct] of LOCALE_ALIASES) {
+  // E.g. things like `en` -> `en-us` or `pt` -> `pt-pt`
+  fixableLocales.set(alias, correct);
+}
+
+// All things like `/en_Us/docs/...` -> `/en-US/docs/...`
+const LOCALE_PATTERNS = [
+  redirect(
+    new RegExp(
+      `^(?<locale>${Array.from(fixableLocales.keys()).join(
+        "|"
+      )})/(?<suffix>.*)`,
```

Comment:
It just occurred to me that we should handle the case when the URL doesn't end with a forward slash (e.g., `/en` --> `/en-US/`, or `/en_us` --> `/en-US/`). Here's the small tweak that does that:
```suggestion
      )})(?:$|/)(?<suffix>.*)`,
```